processing/src/main/java/org/apache/druid/segment/SimpleTopNOptimizationInspector.java

```diff
@@ -19,11 +19,18 @@
 
 package org.apache.druid.segment;
 
-/**
- * @deprecated use {@link Segment} directly as this does nothing
- */
-@Deprecated
-public abstract class AbstractSegment implements Segment
```

Comment:
heh, no idea why git thinks this new file was a rename of this random old file i deleted